src/basins/wada_test.jl

```diff
@@ -0,0 +1,161 @@
+# https://doi.org/10.1016/0020-0190(89)90114-2
+# This compute the Distance transform matrix.
+# Given a matrix, we compute the distance from the 
+# basin with label 0. The result is a matrix whose 
+# entry is the distance to the closest 0 pixel in the 
+# L1 metric (Manhattan). 
+function distance_matrix(basins::BitMatrix) 
+   r,c = size(basins)
+   basdist = ones(Int32,r+2,c+2)*(r+c+4) 
+   # Assign the maximum distance to the pixels not in the basin
+   basdist[2:r+1,2:c+1] .= (1 .- basins) .*(r+c+4) 
+   # first pass right to left, up to bottom
+   for j in 2:c+1, k in 2:r+1
+       basdist[j,k] = w(view(basdist,j-1:j,k-1:k))
+    end       
+   # second pass left to right, bottom up
+   for j in c+1:-1:1, k in r+1:-1:1
+       basdist[j,k] = w2(view(basdist,j:j+1,k:k+1))
+   end       
+   # Remove the extra rows and cols necessary to the alg. 
+   return basdist[2:r+1,2:c+1]
+end
+
+
+"""
+    function haussdorff_distance(M1, M2) -> hd
+
+    Compute the Hausdorff distance between two binary matrices of
+    the same size. First a distance matrix is computed using 
+    `distance_matrix` for each matrix M1 and M2. The entries
+    of this matrix are the distance in L1 metric to the 
+    closest 0 pixel in the initial matrix. The distance being 
+    0 if it belongs to the basin with id 0. 
+"""
+function haussdorff_distance(M1::BitMatrix, M2::BitMatrix)
+    bd1 = distance_matrix(M1) 
+    bd2 = distance_matrix(M2) 
+    hd1 = maximum(bd1.*M2)
+    hd2 = maximum(bd2.*M1)
+    return max(hd1,hd2)
+end
+
+# wada_fractions computes the distance between 
+# boundaries using the hausdorff distance. 
+# The distance is computed using the Shonwilker 
+# algorithm. The result is a matrix with the distance 
+# of the pixels from one boundary to the other. 
+# The algorithm returns the number of pixels
+# that are at a distance strictly more than r
+# (with the hausdorff distance). 
+function wada_fractions(bas1::BitMatrix, bas2::BitMatrix, r::Int)
+    bnd1 = get_boundary(bas1)
+    bd1 = distance_matrix(bnd1) 
+
+    bnd2 = get_boundary(bas2)
+    bd2 = distance_matrix(bnd2) 
+
+    c1 = count(bd1.*bnd2 .> r)
+    c2 = count(bd2.*bnd1 .> r)
+    # @show c1, c2
+    return max(c1,c2)./length(bnd1)
+end
+
+"""
+   merged_boundary(basins, id) -> basins
+
+Returns a matrix with two basins: the first is the basins 
+from the attractor in `id`  and the other basin is formed 
+with the union of all others basins. 
+The function returns a BitMatrix such that the basins of 
+`id` is mapped to 0 and the other basins to 1.  
+""" 
+function merged_basins(basins, id) 
+    mrg_basins = fill!(BitMatrix(undef,size(basins)), false)
+    ids = setdiff(unique(basins), id)
+    for k in ids 
+        I = findall(basins .== k)
+        mrg_basins[I] .= 1
+    end
+    return mrg_basins
+end
+
+"""
+    test_wada_merge(basins, r) -> p
+
+This function test if the 2D array `basins` has the Wada 
+property using the merging technique [Daza2018](@cite). 
```

Comment:
```suggestion
Test if the 2D array `basins` has the Wada 
property using the merging technique of [Daza2018](@cite). 
```

Reply:
Addition, I perhpas it would be useful to cite a paper or Wiki entry that defines the "Wada property".